internal/service/verifier/image_version_verifier.go

```diff
@@ -0,0 +1,110 @@
+package verifier
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kyma-project/modulectl/internal/service/contentprovider"
+)
+
+type Service struct {
+	rawManifestParser RawManifestParser
+}
+
+const managerContainer = "manager"
+
+var (
+	errImageNoTag            = errors.New("no image tag")
+	errNoMatchedVersionFound = errors.New("no matched version found")
+)
+
+type RawManifestParser interface {
+	Parse(filePath string) ([]*unstructured.Unstructured, error)
+}
+
+func NewService(parser RawManifestParser) *Service {
+	return &Service{
+		rawManifestParser: parser,
+	}
+}
+
+func (s *Service) VerifyModuleResources(moduleConfig *contentprovider.ModuleConfig, filePath string) error {
+	resources, err := s.rawManifestParser.Parse(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to parse raw manifest: %w", err)
+	}
+	if moduleConfig.Manager == nil {
+		return nil
+	}
+
+	if err := verifyModuleImageVersion(resources, moduleConfig.Version); err != nil {
+		return fmt.Errorf("failed to verify module image version: %w", err)
+	}
```

Comment:
Currently we check if there is any deployment or stateful set with the given version. With the moduleConfig.Manager, we can actually search the exact deployment by name. You even already checked if moduleConfig.Manager is present.